AdyenEncryption/JOSE/Encryption Algorithms/CommonCryptoHelpers.swift

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import CommonCrypto
+import Foundation
+
+internal func aes256Crypt(operation: CCOperation,
+                          options: CCOptions,
+                          keyData: Data,
+                          initializationVector: Data?,
+                          dataIn: Data) throws -> Data {
+    let dataOutLength = dataIn.count + kCCBlockSizeAES128
+    guard let dataOut = NSMutableData(length: dataOutLength) else {
+        throw JsonWebEncryptionError.encryptionFailed
+    }
+    let keyData = keyData as NSData
+    let dataIn = dataIn as NSData
+    let initializationVector = initializationVector as NSData?
+    var numBytesOut: size_t = 0
+    let algorithm = CCAlgorithm(kCCAlgorithmAES)
+    let status: CCCryptorStatus = CCCrypt(operation,
+                                          algorithm,
+                                          options,
+                                          keyData.bytes, keyData.length,
+                                          initializationVector?.bytes,
+                                          dataIn.bytes, dataIn.length,
+                                          dataOut.mutableBytes, dataOut.length,
+                                          &numBytesOut)
+    guard status == kCCSuccess else {
+        throw JsonWebEncryptionError.encryptionFailed
+    }
+    let startIndex = dataOut.startIndex
+    let endIndex = startIndex.advanced(by: numBytesOut)
+    return (dataOut as Data).subdata(in: 0..<endIndex)
+}
+
+internal func hmac(data: Data, withKey: Data) throws -> Data {
+    guard let dataOut = NSMutableData(length: Int(CC_SHA512_DIGEST_LENGTH)) else {
+        throw JsonWebEncryptionError.unknownError
+    }
+    let dataIn = data as NSData
+    let keyData = withKey as NSData
+    CCHmac(CCHmacAlgorithm(kCCHmacAlgSHA512),
+           keyData.bytes,
+           keyData.length,
+           dataIn.bytes,
+           dataIn.length,
+           dataOut.mutableBytes)
+    return dataOut as Data
+}
+
+internal func generateRandomData(length: Int) throws -> Data {
+    var bytes = [Int8](repeating: 0, count: length)
+    let status = SecRandomCopyBytes(kSecRandomDefault, length, &bytes)
+    guard status == errSecSuccess else { throw JsonWebEncryptionError.failedToGenerateRandomData }
+    return Data(bytes: bytes, count: length)
+}
+
+internal func secKey(fromModulus modulus: String, exponent: String) throws -> SecKey {
+    guard let modulusHex = modulus.hexadecimal,
+          let exponentHex = exponent.hexadecimal else { throw JsonWebEncryptionError.invalidKey }
+    let keyData = generateRSAPublicKey(with: modulusHex, exponent: exponentHex)
+    var error: Unmanaged<CFError>?
+    let parsedKey = SecKeyCreateWithData(keyData as NSData,
```

Comment:
WE have something similar [here](https://github.com/Adyen/adyen-ios/blob/93389d4724ce5d6ed3c75c59b386611e7c1ce86c/AdyenEncryption/Cryptor/RSA.swift#L44)

Reply:
yes, I know, I wanted to start from scratch, hopefully all that old code will be deleted.